From f260539bf50dbf75bfb7a9f25f5500e3b508e858 Mon Sep 17 00:00:00 2001
From: Sasha Levin <sashal@kernel.org>
Date: Wed, 28 Feb 2024 16:50:49 +0100
Subject: fuse: use FUSE_ROOT_ID in fuse_get_root_inode()

From: Miklos Szeredi <mszeredi@redhat.com>

[ Upstream commit 253e52437119719eb87293ef6852e13bb5ad0960 ]

...when calling fuse_iget().

Signed-off-by: Miklos Szeredi <mszeredi@redhat.com>
Stable-dep-of: 928f659a3e36 ("fuse: check for NULL root inode in fuse_fill_super_submount")
Signed-off-by: Sasha Levin <sashal@kernel.org>
---
 fs/fuse/inode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/fuse/inode.c b/fs/fuse/inode.c
index ac7603f2352df..84cfb5fcbc154 100644
--- a/fs/fuse/inode.c
+++ b/fs/fuse/inode.c
@@ -990,7 +990,7 @@ static struct inode *fuse_get_root_inode(struct super_block *sb, unsigned mode)
 	attr.mode = mode;
 	attr.ino = FUSE_ROOT_ID;
 	attr.nlink = 1;
-	return fuse_iget(sb, 1, 0, &attr, 0, 0);
+	return fuse_iget(sb, FUSE_ROOT_ID, 0, &attr, 0, 0);
 }
 
 struct fuse_inode_handle {
-- 
2.53.0

